REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/security/WebSecurity.java

```diff
@@ -0,0 +1,140 @@
+package com.epam.esm.security;
+
+import com.epam.esm.security.exception.RestAccessDeniedHandler;
+import com.epam.esm.security.exception.RestAuthenticationEntryPoint;
+import com.epam.esm.security.jwtconverter.JwtToUserConverter;
+import com.epam.esm.user.entity.roles.Role;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.source.ImmutableJWKSet;
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.proc.SecurityContext;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtEncoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtEncoder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider;
+import org.springframework.security.provisioning.UserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+@Configuration
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@Slf4j
+public class WebSecurity {
+    private final JwtToUserConverter jwtToUserConverter;
+    private final KeyUtils keyUtils;
+    private final PasswordEncoder passwordEncoder;
+    private final UserDetailsManager userDetailsManager;
+    private final RestAuthenticationEntryPoint restAuthenticationEntryPoint;
+    private final RestAccessDeniedHandler restAccessDeniedHandler;
+
+    @Autowired
+    public WebSecurity(JwtToUserConverter jwtToUserConverter,
+                       KeyUtils keyUtils,
+                       PasswordEncoder passwordEncoder,
+                       UserDetailsManager userDetailsManager, RestAuthenticationEntryPoint restAuthenticationEntryPoint, RestAccessDeniedHandler restAccessDeniedHandler) {
```

Comment:
Reformat it please. IMO this looks better
```suggestion
    public WebSecurity(
        JwtToUserConverter jwtToUserConverter,
        KeyUtils keyUtils,
        ....
    )
```

Reply:
done